### ISSUE TITLE
Bikeshed `Container { withVariables, variable, variables }`

### ISSUE THREAD

**Issue by vito** [dagger/dagger#3204]
See discussion here: https://github.com/dagger/dagger/pull/3190#discussion_r984134879

**Comment by dolanor** [comment]
Based on the discussion, it seems the final API would be:

```graphql
extend type Container {
  "Return the value of an env variable"
  envVariable(name: String!): String

  "This container plus a new environment variable set"
  withEnvVariable(name: String!, value: String!): Container!

  "A list of all environment variables"
  envVariables: [EnvVariable]!
}

type EnvVariable {
    name: String!
    value: String!
}
```
Is that correct?
cc: @vito , @shykes , @sipsma

**Comment by dolanor** [comment]
fixed by #3294
